DeviceDetector.php

```diff
@@ -1039,6 +1039,27 @@ protected function parseDevice(): void
             $this->device = AbstractDeviceParser::DEVICE_TYPE_TABLET;
         }
 
+        /**
+         * All devices running Puffin Secure Browser that contain letter 'D' are assumed to be desktops
+         */
+        if (null === $this->device && $this->matchUserAgent('Puffin/.*[LMW]D')) {
```

Comment:
greedy regular expressions should be avoided if possible

'Puffin/(?:\d+[.\d]+)[LMW]D'